spec/system/forgotten_password_spec.rb

```diff
@@ -0,0 +1,53 @@
+require 'rails_helper'
+
+RSpec.describe 'Confirmed user, following forgotten password process', type: :system do
+  let(:user) { create :user, :confirmed }
+  let(:token) { user.send_reset_password_instructions }
+
+  context 'when on the "Choose a password" page' do
```

Comment:
Explicitly referring to the headings can often get messy over time. Better to say something like `when choosing a new password`.